synapse/handlers/message.py

```diff
@@ -1256,12 +1256,22 @@ async def create_new_client_event(
             raise SynapseError(
                 403, "This event is not allowed in this context", Codes.FORBIDDEN
             )
+        # the third-party rules want to replace the event. We'll need to build a new
+        # event.
         elif new_content is not None:
-            # the third-party rules want to replace the event. We'll need to build a new
-            # event.
-            event, context = await self._rebuild_event_after_third_party_rules(
-                new_content, event
-            )
+            if for_batch:
```

Comment:
I was somewhat unsure what to do here-The code for non-batched events here recalculates the state in case the third party rules sent in a state event to the room. I wasn't sure if that was necessary for the batched event, and chose to replace the event using the same prev event ids, state, and auth ids as the batched event that was being replaced, to preserve the batch being persisted in an unbroken chain (ie
creation_event <--- batched_event_1 <----batched_event_2 etc). 
My theory is that the next event sent into the room would resolve the state between the two extremities (the state event the third-party rules sent in and the last event in the batched event chain), but if this is not correct or acceptable the other alternative would be to try and fetch the state event that the third party event sent into the room, add it to the state map of the batched event being rebuilt and recalculate the auth events from that.